pkg/autoscaler/health.go

```diff
@@ -0,0 +1,39 @@
+/*
+Copyright 2016 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package autoscaler
+
+import (
+	"fmt"
+	"net/http"
+
+	"github.com/golang/glog"
+)
+
+func (s *AutoScaler) startHealthz() {
+	http.HandleFunc("/healthz", s.healthFn)
+	glog.Fatal(http.ListenAndServe(":8080", nil))
+}
+
+func (s *AutoScaler) healthFn(w http.ResponseWriter, req *http.Request) {
+	if req.Header.Get("CheckConfigMap") == "" {
```

Comment:
I don't think healthz should be able to cause "real work" to happen.  If we need an external trigger, we should tackle that independently.

Reply:
ah right, it's not idiomatic to do dependency checks via liveness probes. 

We do have a use-case where our we'd like kubelet to restart the autoscaler if it's not able to reach its configmap. I've refactored healthz handler to return the last cached error instead if specified by the `CheckCachedError` header. Would this be an acceptable pattern?